tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(fragmentStoresAndAtomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderTessellationAndGeometryPointSize, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderStorageImageMultisample, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderStorageImageReadWithoutFormat, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderStorageImageArrayDynamicIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderFloat64, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(variableMultisampleRate, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(storagePushConstant16, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(storageInputOutput16, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(multiviewGeometryShader, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(multiviewTessellationShader, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(drawIndirectCount, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(storagePushConstant8, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferInt64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedInt64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderFloat16, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderInputAttachmentArrayDynamicIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderUniformBufferArrayNonUniformIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderInputAttachmentArrayNonUniformIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(descriptorBindingUniformBufferUpdateAfterBind, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(samplerFilterMinmax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(vulkanMemoryModelAvailabilityVisibilityChains, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderOutputViewportIndex, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderOutputLayer, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderDemoteToHelperInvocation, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderTerminateInvocation, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderZeroInitializeWorkgroupMemory, bool, false);
+// Nabla
+NBL_GENERATE_GET_OR_DEFAULT(computeUnits, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(dispatchBase, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(allowCommandBufferQueryCopies, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxOptimallyResidentWorkgroupInvocations, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxResidentInvocations, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(spirvVersion, asset::CGLSLCompiler::E_SPIRV_VERSION, asset::CGLSLCompiler::E_SPIRV_VERSION::ESV_1_6);
+
+// Features Defaults
+// VK 1.0
+NBL_GENERATE_GET_OR_DEFAULT(robustBufferAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(geometryShader, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(tessellationShader, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(depthBounds, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(wideLines, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(largePoints, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(alphaToOne, bool, true);
+NBL_GENERATE_GET_OR_DEFAULT(pipelineStatisticsQuery, bool, false);
```

Comment:
IMO the features should declare BEFORE limits for HLSL traits, but thats a small ordering thing

Reply:
TODO